### DISCUSSION TITLE
Customising PanacheEntityResource

### DISCUSSION THREAD

**Discussion by akil-rails** [quarkusio/quarkus#36878] [Q&A]
Hi

Pls guide on how we can utilise  quarkus-hibernate-reactive-rest-data-panache and also customise the generated methods to always be POST.

The current implementation of quarkus-hibernate-reactive-rest-data-panache is of course rest, meaning there are methods that expect the ID in the resource path, specifically for GET, PUT and DELETE.

We have a situation where we have to use POST for all operations (since the key field is not allowed to be exposed in the URL) - so we need to generate the following paths:

- for GET - POST /get ; the body is { id: "" } instead if the id being in the path param
- for POST - POST /delete ; the body is { id: "" } instead if the id being in the path param
- for PUT - POST / update; the body is the object, and the id is picked up from the body itself.
- for creating the resource, the POST method as generated is suitable

All our resources are like this, [so customising every resource is tedious](https://quarkus.io/guides/rest-data-panache#resource-customisation), and would like to offer something like the following to developers

```
public interface PanResource extends PanacheEntity Rpc Resource<Entity, String> {  }
```

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
@Sgitario mind helping out on this one?

Thanks

**Reply by Sgitario** [reply]
It's not possible to modify the HTTP verb of the generated resources, nor to change how to deal with the request body. 
Where this is technically possible to implement, I'm against doing it since it does not follow the CRUD patterns. 

Instead, I think the best alternative for you is to not use the "quarkus-hibernate-reactive-rest-data-panache" extension, but directly the `quarkus-resteasy-reactive` and the Panache extensions and provide a custom resource that implements the resources you need, so developers can extend it to.
